src/main/java/org/ehrbase/fhirbridge/ehr/converter/specific/virologischerbefund/BefundJedesEreignisPointEventConverter.java

```diff
@@ -0,0 +1,94 @@
+import org.ehrbase.fhirbridge.ehr.opt.virologischerbefundcomposition.definition.ProAnalytCluster;
+import org.ehrbase.fhirbridge.ehr.opt.virologischerbefundcomposition.definition.ProbeCluster;
+import org.ehrbase.fhirbridge.ehr.opt.virologischerbefundcomposition.definition.AnatomischeLokalisationCluster;
+import org.ehrbase.fhirbridge.ehr.opt.virologischerbefundcomposition.definition.NameDerKoerperstelleDefiningCode;
+import org.ehrbase.fhirbridge.ehr.opt.virologischerbefundcomposition.definition.LabortestBezeichnungDefiningCode;
+import org.ehrbase.fhirbridge.ehr.converter.generic.TimeConverter;
+
+import org.hl7.fhir.exceptions.FHIRException;
+import org.hl7.fhir.r4.model.Coding;
+import org.hl7.fhir.r4.model.Observation;
+import org.hl7.fhir.r4.model.Specimen;
+
+import java.time.temporal.TemporalAccessor;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+public class BefundJedesEreignisPointEventConverter extends ObservationToPointEventConverter<BefundJedesEreignisPointEvent> {
+
+    @Override
+    protected BefundJedesEreignisPointEvent convertInternal(Observation observation) throws FHIRException {
+
+        Specimen specimen = observation.getSpecimenTarget();
+        BefundJedesEreignisPointEvent befundevent = new BefundJedesEreignisPointEvent();
+
+        if(checkLabortestbezeichnung(observation)){
+            befundevent.setLabortestBezeichnungDefiningCode(LabortestBezeichnungDefiningCode.DETECTION_OF_VIRUS_PROCEDURE);
+        }else {
+            throw new ConversionException("createLabortestBezeichnungDefiningCode failed as snomedct-subcategory Code was not 122442008.");
+        }
+        mapProbe(specimen).ifPresent(befundevent::setProbe);
+
+        List<ProAnalytCluster> proAnalytClusterlist = new ArrayList<>();
+        LabortestPanelCluster labortestPanelCluster = new LabortestPanelCluster();
+
+        proAnalytClusterlist.add(new ProAnalytClusterConverter().convert(observation));
+        labortestPanelCluster.setProAnalyt(proAnalytClusterlist);
+        befundevent.setLabortestPanel(labortestPanelCluster);
+
+        return befundevent;
+
+    }
+
+    private boolean checkLabortestbezeichnung(Observation observation) {
+        for (Coding loop : observation.getCategory().get(0).getCoding()){
+            if(loop.getCode().equals("122442008")) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    private Optional<ProbeCluster> mapProbe(Specimen specimen) throws FHIRException {
+        ProbeCluster probecluster = new ProbeCluster();
+        if (specimen.hasCollection()){
+            mapZeitpunktDerProbenentnahme(specimen).ifPresent(probecluster::setZeitpunktDerProbenentnahmeValue);
+
+            AnatomischeLokalisationCluster anatomischeLokalisationCluster = new AnatomischeLokalisationCluster();
+            for (Coding loop : specimen.getCollection().getBodySite().getCoding()){
```

Comment:
Is now checked as well in an extra methods. "2 methods" as otherwise the if/for checks lead to a code smell.
https://github.com/ehrbase/fhir-bridge/pull/392/files#diff-7e8996e1193679795973b45c1d8e9b897e57859669d2d6a01138153dfe06ad49